code/modules/projectiles/projectile/magic/beam_spells.dm

```diff
@@ -0,0 +1,25 @@
+/obj/projectile/magic/ray
+	hitscan = TRUE
+
+/obj/projectile/magic/ray/frost
+	name = "ray of frost"
+	icon_state = "ice_2"
+	damage = 8
+	damage_type = BURN
+	nodamage = FALSE
+	tracer_type = /obj/effect/projectile/tracer/laser/blue
+	muzzle_type = /obj/effect/projectile/muzzle/laser/blue
+	impact_type = /obj/effect/projectile/impact/laser/blue
+
+/obj/projectile/magic/ray/frost/on_hit(atom/target, blocked = 0)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		L.bodytemperature -= 100
```

Comment:
Use min() or people could potentially get to impossible temps.
Or better yet, use a sanitized proc. There’s probably one